### PR TITLE
Issue/4410 offline messaging

### LINKED ISSUE #4410: Improve errors when the app is offline
### Expected behavior

If the app is offline and can't sync/display the correct information on a screen, I expect to see a toast message or other error message explaining that.

In most cases, when an action fails while offline, we display this toast message:

> There is no network available
### Actual behavior

There are a couple screens where there is no error or the error is generic, not explaining the offline state:
- Plans: A toast message appears `Unable to load plans`
- Themes: When you first open the Themes section, the full list of themes does not appear and there is no error or other message to explain why
- Editor - Post Settings - Select categories: A blank screen appears (no categories load) and there is no error or other message to explain why
- Editor - Post Settings - Add a new category: A toast message appears `Adding category failed` with no explanation why
- Me - My Profile: It looks like the changes are saved (no error message), but if you exit the My Profile section and open it again the changes are gone.
#### Screenshots

Examples of clear toast/error messages:

screenshot_20160721-122502

screenshot_20160721-124101

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #4410 - The following changes have been made when no network is available:

* Plans: opening plans shows a "no connection" toast now
* Editor > Post Settings > Select categories: category list now shows a "no connection" empty view
* Editor > Post Settings > Add category:  "no connection" toast appears and add dialog isn't shown
* Me > My Profile: "no connection" toast appears after user edits any profile information

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/SelectCategoriesActivity.java [hunk 1]
```diff
@@ -118,9 +123,14 @@ public void onRefreshStarted() {
...
         populateCategoryList();
 
-        // Refresh blog list if network is available and activity really starts
-        if (NetworkUtils.isNetworkAvailable(this) && savedInstanceState == null) {
-            refreshCategories();
+        if (NetworkUtils.isNetworkAvailable(this)) {
+            mEmptyView.setText(R.string.empty_list_default);
+            // Refresh blog list if network is available and activity really starts
+            if (savedInstanceState == null) {
+                refreshCategories();
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/SelectCategoriesActivity.java]: Just want to confirm one thing, If you open this activity while having no internet, then connect to internet and rotate the screen, this method wont be reached (savedInstanceState wont be null). Is ok?

File: WordPress/src/main/res/layout/select_categories.xml [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:padding="0px">
 
-    <org.wordpress.android.util.widgets.CustomSwipeRefreshLayout
-        xmlns:android="http://schemas.android.com/apk/res/android"
+    <org.wordpress.android.util.widgets.CustomSwipeRefreshLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (khaykov) [inline WordPress/src/main/res/layout/select_categories.xml]: I think we can safely remove second `xmlns:android` attribute from here.

File: WordPress/src/main/res/layout/select_categories.xml [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    android:layout_width="fill_parent"
-    android:layout_height="fill_parent"
-    android:padding="0px"
-    android:orientation="vertical">
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
```
Comment (khaykov) [inline WordPress/src/main/res/layout/select_categories.xml]: We switched from LinearLayout to FrameLayout, so we don't need `android:orientation` anymore.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/SelectCategoriesActivity.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/SelectCategoriesActivity.java]: > If you open this activity while having no internet, then connect to internet and rotate the screen, this method wont be reached

Good point - fixed in 1b29112

File: WordPress/src/main/res/layout/select_categories.xml (hunk 3 above)
Comment (nbradbury) [inline WordPress/src/main/res/layout/select_categories.xml]: Corrected in cfed579

**Comment by nbradbury** [comment]
@khaykov Ready for another round, thanks.

**Comment by khaykov** [comment]
Thanks for the fixes, Nick! :shipit:

**Pull request merged**